src/modROS.lua

```diff
@@ -491,6 +487,13 @@ function ModROS:rosPubMsg(flag)
             end
         end
 
+        -- initialise publishers
+        self._pub_tf = Publisher.new(self.file_pipe, "tf", tf2_msgs_TFMessage)
+        self._pub_clock = Publisher.new(self.file_pipe, "clock", rosgraph_msgs_Clock)
+        self._pub_odom = Publisher.new(self.file_pipe, "odom", nav_msgs_Odometry)
+        self._pub_scan = Publisher.new(self.file_pipe, "scan", sensor_msgs_LaserScan)
+        self._pub_imu = Publisher.new(self.file_pipe, "imu", sensor_msgs_Imu)
```

Comment:
I'm not too happy with all these lines here, but it's consistent with what we have now and `file_pipe` is only non-`nil` here.

If/when we decouple opening `file_pipe` from `rosPubMsg(..)`, we can change this to something nicer.

Reply:
I've since rewritten this and this comment is no longer relevant.